app/controllers/RouterController.scala

```diff
@@ -57,33 +57,41 @@ trait RouterController extends FrontendController with Actions {
 
   def auditConnector: AuditConnector
 
-  def createAuditContext(): TAuditContext
-
   def analyticsEventSender: AnalyticsEventSender
 
-  val account = AuthenticatedBy(authenticationProvider = RouterAuthenticationProvider, pageVisibility = AllowAll).async { implicit user => request => route(user, request) }
+  val account = AuthenticatedBy(authenticationProvider = RouterAuthenticationProvider, pageVisibility = AllowAll).async {
+    implicit user =>
+      request =>
+        route(user, request)
+  }
 
   def route(implicit authContext: AuthContext, request: Request[AnyContent]): Future[Result] = {
     val ruleContext = RuleContext(None)
-    val auditContext = createAuditContext()
-    getFinalDestination(ruleContext, auditContext).map(location => Redirect(location.fullUrl))
-  }
 
-  private def getFinalDestination(ruleContext: RuleContext, auditContext: TAuditContext)(implicit request: Request[AnyContent], authContext: AuthContext) = for {
-    destinationAfterRulesApplied <- ruleEngine.matchRulesForLocation(ruleContext, auditContext)
-    finalDestination <- throttlingService.throttle(destinationAfterRulesApplied, auditContext, ruleContext)
-  } yield {
-    sendAuditEvent(auditContext, finalDestination)
-    metricsMonitoringService.sendMonitoringEvents(auditContext, finalDestination)
-    Logger.debug(s"routing to: ${finalDestination.name}")
-    analyticsEventSender.sendEvents(finalDestination.name, auditContext)
-    finalDestination
-  }
-
-  private def sendAuditEvent(auditContext: TAuditContext, throttledLocation: Location)(implicit authContext: AuthContext, request: Request[AnyContent], hc: HeaderCarrier) = {
-    auditContext.toAuditEvent(throttledLocation).foreach { auditEvent =>
+    def sendAuditEvent(auditInfo: AuditInfo, throttledLocation: Location)(implicit authContext: AuthContext, request: Request[AnyContent], hc: HeaderCarrier) = {
+      val auditEvent = auditInfo.toAuditEvent(throttledLocation)
       auditConnector.sendEvent(auditEvent)
-      Logger.debug(s"Routing decision summary: ${auditEvent.detail \ "reasons"}")
+      val reasons: JsValue = (auditEvent.detail \ "reasons").getOrElse(JsNull)
+      Logger.debug(s"Routing decision summary: ${Json.stringify(reasons)}")
+    }
+
+    val destinationAfterRulesApplied = ruleEngine.getLocation(ruleContext)
+    val finalResult = throttlingService.throttle(destinationAfterRulesApplied, ruleContext).run
+    val futureAuditInfo = finalResult map { case (auditInfo, _) => auditInfo }
+    val futureFinalDestination = finalResult map { case (_, finalDestination) => finalDestination }
+
+    val destination = for {
+      finalDestination <- futureFinalDestination
+      auditInfo <- futureAuditInfo
+    } yield {
+      sendAuditEvent(auditInfo, finalDestination)
+      metricsMonitoringService.sendMonitoringEvents(auditInfo, finalDestination)
+      analyticsEventSender.sendEvents(finalDestination.name, auditInfo)
```

Comment:
the two lines above this take `auditInfo` and `finalDestination` in the opposite order. can we standardise this one way or the other?